frappe/utils/print_format.py

```diff
@@ -112,8 +113,21 @@ def read_multi_pdf(output):
 	return filedata
 
 
-@frappe.whitelist()
+@frappe.whitelist(allow_guest=True)
 def download_pdf(doctype, name, format=None, doc=None, no_letterhead=0):
+	doc = frappe.get_doc(doctype, name)
+	doc.doctype = doctype
+	try:
+		validate_print_permission(doc)
+	except frappe.exceptions.LinkExpiredError:
+		frappe.local.response.http_status_code = 410
+		frappe.local.response.message = _("Link Expired")
+		return
+	except frappe.exceptions.InvalidKey:
+		frappe.local.response.http_status_code = 401
+		frappe.local.response.message = _("Invalid Key")
+		return
```

Comment:
This is mostly to avoid bad UX... without it, the PDF generated in case of exception will be like
![Screenshot 2022-06-01 at 10 26 17 AM](https://user-images.githubusercontent.com/13928957/171330900-3c474b91-32f5-4e8d-9cc1-1667f746e5e7.png)